superset-frontend/src/SqlLab/components/SaveDatasetModal/index.tsx

```diff
@@ -17,153 +17,362 @@
   }
-  .smd-input {
+  .sdm-input {
     margin-left: 45px;
     width: 401px;
   }
-  .smd-autocomplete {
+  .sdm-autocomplete {
     margin-left: 8px;
     width: 401px;
   }
-  .smd-radio {
+  .sdm-radio {
     display: block;
     height: 30px;
     margin: 10px 0px;
     line-height: 30px;
   }
-  .smd-overwrite-msg {
+  .sdm-overwrite-msg {
     margin: 7px;
   }
 `;
 
+const updateDataset = async (
+  dbId: number,
+  datasetId: number,
+  sql: string,
+  columns: Array<Record<string, any>>,
+  owners: [number],
+  overrideColumns: boolean,
+) => {
+  const endpoint = `api/v1/dataset/${datasetId}?override_columns=${overrideColumns}`;
+  const headers = { 'Content-Type': 'application/json' };
+  const body = JSON.stringify({
+    sql,
+    columns,
+    owners,
+    database_id: dbId,
+  });
+
+  const data: JsonResponse = await SupersetClient.put({
+    endpoint,
+    headers,
+    body,
+  });
+  return data.json.result;
+};
+
+// const getDatasourceState = (datasource: ExploreDatasource) => {
+//   if (datasource.type === DatasourceType.Dataset) return datasource as Dataset;
+//   return datasource as Query;
+// };
+
 // eslint-disable-next-line no-empty-pattern
 export const SaveDatasetModal: FunctionComponent<SaveDatasetModalProps> = ({
   visible,
-  onOk,
   onHide,
-  handleDatasetNameChange,
-  handleSaveDatasetRadioBtnState,
-  saveDatasetRadioBtnState,
-  shouldOverwriteDataset,
-  handleOverwriteCancel,
-  handleOverwriteDataset,
-  handleOverwriteDatasetOption,
-  defaultCreateDatasetValue,
-  disableSaveAndExploreBtn,
-  handleSaveDatasetModalSearch,
-  filterAutocompleteOption,
-  userDatasetOptions,
-  onChangeAutoComplete,
-}) => (
-  <StyledModal
-    show={visible}
-    title="Save or Overwrite Dataset"
-    onHide={onHide}
-    footer={
-      <>
-        {!shouldOverwriteDataset && (
-          <Button
-            disabled={disableSaveAndExploreBtn}
-            buttonStyle="primary"
-            onClick={onOk}
-          >
-            {t('Save & Explore')}
-          </Button>
-        )}
-        {shouldOverwriteDataset && (
-          <>
-            <Button onClick={handleOverwriteCancel}>Back</Button>
+  buttonTextOnSave,
+  buttonTextOnOverwrite,
+  modalDescription,
+  datasource,
+}) => {
+  const getDefaultDatasetName = () =>
+    `${datasource.tab} ${moment().format('MM/DD/YYYY HH:mm:ss')}`;
+  const [datasetName, setDatasetName] = useState(getDefaultDatasetName());
+  const [newOrOverwrite, setNewOrOverwrite] = useState(
+    DatasetRadioState.SAVE_NEW,
+  );
+  const [shouldOverwriteDataset, setShouldOverwriteDataset] = useState(false);
+  const [userDatasetOptions, setUserDatasetOptions] = useState<
+    DatasetOptionAutocomplete[]
+  >([]);
+  const [datasetToOverwrite, setDatasetToOverwrite] = useState<
+    Record<string, any>
+  >({});
+  const [autocompleteValue, setAutocompleteValue] = useState('');
+
+  const user = useSelector<SqlLabExploreRootState, User>(state =>
+    getInitialState(state),
```

Comment:
is this returning the user or the entire state?

Reply:
Good call, I clarified the parameter name in [`this commit`](https://github.com/apache/superset/pull/19855/commits/419fbb7b6de02e6a5cc0efeb16d840861b165fad).